src/errors.rs

```diff
@@ -12,6 +14,14 @@ error_type! {
     pub enum SpaceapiServerError {
         Redis(RedisError) {
             cause;
-        }
+        },
+        IoError(io::Error) {
+            cause;
+        },
+        Message(Cow<'static, str>) {
+            desc (e) &**e;
+            from (s: &'static str) s.into();
+            from (s: String) s.into();
+        },
```

Comment:
> And it's caused by

Ah, that's what I was asking for. But in essence both errors are the same. They both point at problems with processing the socket address.

We could probably create an error type that handles both cases, but probably not through the rust-error-type macro. But I'm not sure if we should continue using the macro anyways, it takes away some flexibility. See also: https://github.com/DanielKeep/rust-error-type/issues/2 What do you think, @rnestler?

If we just go with the generic string message, it won't be possible for user code to differentiate between argument problems and other problems that are returned as strings.